sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/configuration/StatsbeatConnectionString.java

```diff
@@ -111,8 +113,8 @@ public static StatsbeatConnectionString create(
             ingestionEndpoint += "/";
         }
         try {
-            endpointUrl = new URL(ingestionEndpoint);
-        } catch (MalformedURLException e) {
+            endpointUrl = new URI(ingestionEndpoint).toURL();
```

Comment:
what was the warning about using URL constructor? wonder if it applies to URL in general?

Reply:
deprecation